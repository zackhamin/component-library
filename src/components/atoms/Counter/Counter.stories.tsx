import { Story } from '@storybook/react';
import { useState } from 'react';

import { FormControl } from 'Molecules/FormControl';
import { FormLabel } from 'Atoms/FormLabel';
import { TextInput } from 'Atoms/TextInput';
import { Counter, CounterProps } from './Counter';

export default {
	title: 'Atoms/Counter',
	component: Counter,
	args: {
		current: 0,
		max: 10,
		fontSize: '1.6rem',
	},
};

const Template: Story<CounterProps> = (args) => <Counter {...args} />;

const InputTemplate: Story<CounterProps> = (args) => {
	const [inputValue, setInputValue] = useState('');

	return (
		<FormControl id="testId" name="testName">
			<div
				style={{
					display: 'flex',
					justifyContent: 'space-between',
					alignItems: 'baseline',
				}}
			>
				<FormLabel>Label</FormLabel>
				<Counter current={inputValue.length} max={args.max} />
			</div>
			<TextInput onChange={(e) => setInputValue(e.target.value)} />
		</FormControl>
	);
};

export const Default = Template.bind({});

export const NoMaxValue = Template.bind({});
NoMaxValue.args = {
	max: undefined,
};

export const Input = InputTemplate.bind({});
Input.args = {
	max: 100,
};

export const InputWithNoMaxValue = InputTemplate.bind({});
InputWithNoMaxValue.args = {
	max: undefined,
};
