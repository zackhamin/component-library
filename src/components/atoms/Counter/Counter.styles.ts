import { colors } from 'Styles';
import { CounterProps } from './Counter';
import styled from 'styled-components';

const StyledCounter = styled.div<CounterProps>`
	font-size: ${({ fontSize }) => fontSize};
	color: ${({ current, max }) => {
		if (max && current > max) {
			return colors.error;
		}
		return colors.grey500;
	}};

	span {
		color: ${colors.offBlack};
	}
`;

export { StyledCounter };
