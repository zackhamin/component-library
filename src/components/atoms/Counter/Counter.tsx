import { Size } from '@Types/css';
import { StyledCounter } from './Counter.styles';

export interface CounterProps {
	current: number;
	max?: number;
	fontSize?: Size;
}

const Counter = ({ current, max, fontSize = '1.6rem' }: CounterProps) => {
	return (
		<StyledCounter fontSize={fontSize} current={current} max={max}>
			{current}
			{!!max && <span>/{max}</span>}
		</StyledCounter>
	);
};

export { Counter };
