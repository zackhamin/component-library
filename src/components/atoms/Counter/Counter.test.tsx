import { colors } from 'Styles';
import { Counter } from './Counter';
import { render } from '@testing-library/react';

describe('Counter', () => {
	it('renders correctly', () => {
		const { container } = render(<Counter current={0} max={100} />);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('shows current value in red if current is larger than max', () => {
		const { container } = render(<Counter current={101} max={100} />);

		expect(container.firstChild).toHaveStyle(`color: ${colors.error}`);
	});
});
