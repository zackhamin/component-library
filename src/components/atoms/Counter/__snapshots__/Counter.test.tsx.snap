// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #979797;
}

.c0 span {
  color: #333333;
}

<div
  class="c0"
  font-size="1.6rem"
  max="100"
>
  0
  <span>
    /
    100
  </span>
</div>
`;
